examples/cms/src/App.tsx

```diff
@@ -0,0 +1,87 @@
+import "./App.css"
+
+import type { ManagedCollection } from "framer-plugin"
+
+import { framer } from "framer-plugin"
+import { useEffect, useLayoutEffect, useState } from "react"
+import { DataSource, getDataSource, getDataSourcesIds } from "./data"
+import { FieldMapping } from "./FieldMapping"
+import { SelectDataSource } from "./SelectDataSource"
+import { Spinner } from "./components/Spinner"
+
+interface AppProps {
+    collection: ManagedCollection
+    previouslyConfiguredDataSourceId: string | null
+    previouslyConfiguredSlugFieldId: string | null
+}
+
+export function App({ collection, previouslyConfiguredDataSourceId, previouslyConfiguredSlugFieldId }: AppProps) {
+    const [dataSource, setDataSource] = useState<DataSource | null>(null)
+
+    const [isLoadingConfiguredDataSource, setIsLoadingConfiguredDataSource] = useState(
+        Boolean(previouslyConfiguredDataSourceId)
+    )
+    const hasDataSourceSelected = Boolean(isLoadingConfiguredDataSource || dataSource)
+
+    useLayoutEffect(() => {
+        framer.showUI({
+            width: hasDataSourceSelected ? 360 : 320,
+            height: hasDataSourceSelected ? 425 : 305,
+            minWidth: hasDataSourceSelected ? 360 : undefined,
+            minHeight: hasDataSourceSelected ? 425 : undefined,
+            resizable: dataSource !== null,
+        })
+    }, [hasDataSourceSelected, dataSource])
+
+    useEffect(() => {
+        if (!previouslyConfiguredDataSourceId) {
+            return
+        }
+
+        const abortController = new AbortController()
+
+        setIsLoadingConfiguredDataSource(true)
+        getDataSource(previouslyConfiguredDataSourceId, abortController.signal)
+            .then(setDataSource)
+            .catch(error => {
+                if (abortController.signal.aborted) {
+                    return
+                }
+
+                console.error(error)
+                framer.notify(
+                    `Error loading previously configured data source "${previouslyConfiguredDataSourceId}". Check the logs for more details.`,
+                    {
+                        variant: "error",
+                    }
+                )
+            })
+            .finally(() => {
+                if (abortController.signal.aborted) {
+                    return
+                }
+
+                setIsLoadingConfiguredDataSource(false)
+            })
+
+        return () => {
+            abortController.abort()
+        }
+    }, [previouslyConfiguredDataSourceId])
+
+    if (isLoadingConfiguredDataSource) {
+        return <Spinner />
+    }
+
+    if (!dataSource) {
+        return <SelectDataSource dataSourceIds={getDataSourcesIds()} onSelectDataSource={setDataSource} />
```

Comment:
I would call `getDataSourcesIds` within the component `SelectDataSource`. No reason to make something static an API

Reply:
Als naming wise I would only make the last word plural. So:
`getDataSourcesIds` → `getDataSourceIds`